background.ts

```diff
@@ -247,18 +247,17 @@ Electron.app.whenReady().then(async() => {
       }
       console.log(`Failed to update command from argument ${ commandLineArgs.join(', ') }`, err);
     }
+
     httpCommandServer = new HttpCommandServer(new BackgroundCommandWorker());
     await httpCommandServer.init();
     await httpCredentialHelperServer.init();
 
+    await initUI();
+    await checkForBackendLock();
```

Comment:
We're doing all of first run then starting the app, _then_ checking for the lock? That seems odd?

Note that at this point I'm not sure I understand the flow around here correctly, so consider this more of a request to double check, and not saying that this is wrong.

Reply:
I was just unsure if modifying the settings file (as part of first run) during a snapshot could lead to issues. The bulk of the snapshot would be the VM, of course, so that's _probably_ fine…